### PR TITLE
Replace null checks with IsEmpty property for ReadOnlySpan<char>

### PR DESCRIPTION
**Pull request by prozolic**
This PR replace null checks with `IsEmpty` property for `ReadOnlySpan<char>`. 
The change suppresses [CA2265](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2265) warnings.

### CODE REVIEW

File: src/Markdig/Helpers/LinkHelper.cs [hunk 1]
```diff
@@ -60,12 +60,12 @@ public static string Urilize(ReadOnlySpan<char> headingText, bool allowOnlyAscii
             }
             else
             {
-                normalized = allowOnlyAscii ? CharNormalizer.ConvertToAscii(c) : null;
+                normalized = allowOnlyAscii ? CharNormalizer.ConvertToAscii(c) : ReadOnlySpan<char>.Empty;
             }
 
             for (int j = 0; j < (normalized.Length < 1 ? 1 : normalized.Length); j++)
```
Comment (MihaZupan) [inline src/Markdig/Helpers/LinkHelper.cs:66]: Are we missing test coverage given that a null `normalized` would throw here?

Comment (prozolic) [reply src/Markdig/Helpers/LinkHelper.cs:66]: @MihaZupan Thank you for the review.
Since `normalized` is a `ReadOnlySpan<char>`, I changed the check from `normalized != null` to `!normalized.IsEmpty`.
When `CharNormalizer.ConvertToAscii(c)` returns `null`, it is implicitly converted to an empty span, maintaining the same behavior as before.
I confirmed from the following test cases.

https://github.com/xoofx/markdig/blob/master/src/Markdig.Tests/TestLinkHelper.cs#L330-L337

Comment (MihaZupan) [reply src/Markdig/Helpers/LinkHelper.cs:66]: Right, I was more referring to whether there are cases that could trigger an exception to be thrown before your current change. If so, it'd be good to have a test case for it.

Comment (prozolic) [reply src/Markdig/Helpers/LinkHelper.cs:66]: At this point, I cannot find any cases where an exception would be thrown before my change. 
If an empty string (`""`) were assigned to `normalized`, the results of `normalized != null` and `!normalized.IsEmpty` would differ, which could cause an exception. However, I cannot find any cases where `CharNormalizer.ConvertToAscii` would return an empty string (`""`).
Does this response address your concern?
